src/main/java/org/jabref/logic/citationstyle/CSLAdapter.java

```diff
@@ -0,0 +1,116 @@
+package org.jabref.logic.citationstyle;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jabref.model.entry.BibEntry;
+
+import de.undercouch.citeproc.CSL;
+import de.undercouch.citeproc.ItemDataProvider;
+import de.undercouch.citeproc.ListItemDataProvider;
+import de.undercouch.citeproc.bibtex.BibTeXConverter;
+import de.undercouch.citeproc.csl.CSLItemData;
+import de.undercouch.citeproc.output.Bibliography;
+import org.jbibtex.BibTeXEntry;
+import org.jbibtex.DigitStringValue;
+import org.jbibtex.Key;
+
+/**
+ * Provides an adapter class to CSL. It holds a CSL instance under the hood that is only recreated when
+ * the style changes.
+ *
+ * @apiNote The first call to {@link #makeBibliography} is expensive since the
+ * CSL instance will be created. As long as the style stays the same, we can reuse this instance. On style-change, the
+ * engine is re-instantiated. Therefore, the use-case of this class is many calls to {@link #makeBibliography} with the
+ * same style. Changing the output format is cheap.
+ * @implNote The main function {@link #makeBibliography} will enforce
+ * synchronized calling. The main CSL engine under the hood is not thread-safe. Since this class is usually called from
+ * a SwingWorker, the only other option would be to create several CSL instances which is wasting a lot of resources and very slow.
+ * In the current scheme, {@link #makeBibliography} can be called as usual
+ * SwingWorker task and to the best of my knowledge, concurrent calls will pile up and processed sequentially.
+ */
+public class CSLAdapter {
+
+    private static final BibTeXConverter BIBTEX_CONVERTER = new BibTeXConverter();
+    private static String style;
+    private static CitationStyleOutputFormat format;
+    private static JabRefItemDataProvider dataProvider = new JabRefItemDataProvider();
+    private static CSL cslInstance;
```

Comment:
Is it needed, that everything is `static` here? I actually would prefer a usual class with non-standard methods/instance variables.

Reply:
Yes. This would probably be better. We could then still instantiate several CSL adapters if we need one for whatever purpose.